Support no auto encode for uri parameters

twoplustwoone opened the pull request.

twoplustwoone committed:
feat: add support for no auto encode

twoplustwoone committed:
chore: update package lock

twoplustwoone committed:
chore: update deps and update models

twoplustwoone committed:
fix: change to noAutoEncode

Negative is better for backwards compatibility

twoplustwoone committed:
chore: update travis.yml

jarrodek reviewed (changes requested):
This one cannot go without tests. It's a core logic change.

twoplustwoone committed:
fix(uri): read noAutoEncodes correctly

Change the iterattion of the map entries to an array
This is to have access to index

jarrodek reviewed (changes requested):
LGTM
